src/compression.rs

```diff
@@ -153,6 +153,9 @@ generate! {
         "lz4" => Lz4("lz4", lz4_flex::frame::FrameDecoder<R>) {
             lz4_flex::frame::FrameDecoder::new(r)
         },
+        "zst" => Zst("zstd/Zstandard", zstd::stream::read::Decoder<'static, R>) {
+            zstd::stream::read::Decoder::with_buffer(r).unwrap()
```

Comment:
You should be able to do this in case opening the file fails:
```suggestion
            zstd::stream::read::Decoder::with_buffer(r)?
```